File: L014/palinSubstr.java
import java.util.Scanner;

public class palinSubstr{


    public static Scanner sc = new Scanner(System.in);

    public static void palinSubstr_(String str){

        for(int i=0;i<str.length();i++){
            for(int len=1;i+len<=str.length();len++){
                String s = str.substring(i,i+len);
                if(isPalindrome(s)){
                    System.out.println(s);
                }
            }
        }

    }

    public static boolean isPalindrome(String str){
        int i=0;
        int j=str.length()-1;

    while(i<j){

        if(str.charAt(i++) != str.charAt(j--)) return false;

    }
    return true;

    }
    
    public static void main(String[] args){

        String str = sc.nextLine();

        palinSubstr_(str);
    }
}

File: L014/palinSubstr.cpp
#include<iostream>
using namespace std;

bool isPalindrome(string str){

    int i=0;
    int j=str.length()-1;

    while(i<j){

        if(str[i++] != str[j--]) return false;

    }
    return true;
}
void palinSubstr(string str){

    for(int i=0;i<str.length();i++){
            for(int len=1;i+len<=str.length();len++){
                
                string s = str.substr(i,len);
                if(isPalindrome(s)){
                    cout<<s<<endl;
                }
            }
        }
}



int main(){

    string str;
    cin>>str;

    palinSubstr(str);
    return 0;
}
